src/backend/InvenTree/plugin/installer.py

```diff
@@ -63,39 +66,54 @@ def handle_pip_error(error, path: str) -> list:
         raise ValidationError(errors[0])
 
 
-def check_package_path(packagename: str):
-    """Determine the install path of a particular package.
+def get_install_info(packagename: str) -> dict:
+    """Determine the install information for a particular package.
 
-    - If installed, return the installation path
-    - If not installed, return False
+    - Uses 'pip show' to determine the install location of a package.
     """
-    logger.debug('check_package_path: %s', packagename)
+    logger.debug('get_install_info: %s', packagename)
 
     # Remove version information
-    for c in '<>=! ':
+    for c in '<>=!@ ':
         packagename = packagename.split(c)[0]
 
+    info = {}
+
     try:
         result = pip_command('show', packagename)
 
         output = result.decode('utf-8').split('\n')
 
         for line in output:
-            # Check if line matches pattern "Location: ..."
-            match = re.match(r'^Location:\s+(.+)$', line.strip())
+            parts = line.split(':')
+
+            if len(parts) >= 2:
+                key = str(parts[0].strip().lower().replace('-', '_'))
+                value = str(parts[1].strip())
 
-            if match:
-                return match.group(1)
+                info[key] = value
 
     except subprocess.CalledProcessError as error:
-        log_error('check_package_path')
+        log_error('get_install_info')
 
         output = error.output.decode('utf-8')
+        info['error'] = output
         logger.exception('Plugin lookup failed: %s', str(output))
-        return False
 
-    # If we get here, the package is not installed
-    return False
+    return info
+
+
+def plugins_file_hash():
+    """Return the file hash for the plugins file."""
+    import hashlib
+
+    pf = settings.PLUGIN_FILE
+
+    if not pf or not pf.exists():
+        return None
+
+    with pf.open('rb') as f:
+        return hashlib.md5(f.read()).hexdigest()
```

Comment:
Q (non-blocking): Can we use this oportunity to move to a sha hashing function? md5 always comes up in security reviews and toolings.
```suggestion
        return hashlib.sha256(f.read()).hexdigest()
```